cluster.go

```diff
@@ -1172,11 +1172,12 @@ func (c *Cluster) globalPinInfoCid(method string, h cid.Cid) (api.GlobalPinInfo,
 		if r.Status == api.TrackerStatusBug {
 			logger.Errorf("%s: error in broadcast response from %s: %s ", c.id, members[i], e)
 			pin.PeerMap[members[i]] = api.PinInfo{
-				Cid:    h,
-				Peer:   members[i],
-				Status: api.TrackerStatusClusterError,
-				TS:     time.Now(),
-				Error:  e.Error(),
+				Cid:      h,
+				Peer:     members[i],
+				PeerName: string(members[i]),
```

Comment:
This should be `members[i].String()` or ` members[i].Pretty()` (leaning towards the first) . Unfortunately casting a peer.ID to string directly is a bad thing to do (been there, done that).

Reply:
Makes sense. I was able to find a nice article explaining why.
https://stackoverflow.com/questions/4167304/why-should-casting-be-avoided

Reply:
Actually, the reason here is that a peer ID is internally a `string` but that string is just holding a byte slice (a multihash actually) and the string representation we use for peer IDs is the B58 encoded version of that byte slice. https://godoc.org/github.com/libp2p/go-libp2p-peer#IDB58Encode . The reason why a Peer ID is a `string` and not a `[]byte` probably comes to the slightly smaller footprint and immutability of strings (https://syslog.ravelin.com/byte-vs-string-in-go-d645b67ca7ff). In any case, the right way to get a string is to IDB58Encode it.